src/traits/mod.rs

```diff
@@ -45,7 +46,13 @@ pub trait Group:
   type CompressedGroupElement: CompressedGroup<GroupElement = Self>;
 
   /// A type representing preprocessed group element
-  type PreprocessedGroupElement: Clone + Debug + Send + Sync + Serialize + for<'de> Deserialize<'de>;
+  type PreprocessedGroupElement: CurveAffine
```

Comment:
This is probably worth linking to #198

Reply:
Hi, @ashWhiteHat could we avoid adding `CurveAffine` requirement to this type here? This introduces a dependency on group crate from any curve library that can be used to implement a provider. What is the gain from adding this requirement?

Reply:
Thank you for the question.
That's necessary for test abstraction of [test_from_label](https://github.com/microsoft/Nova/blob/main/src/provider/bn256_grumpkin.rs#L92).

It checks return value of [from_label](https://github.com/microsoft/Nova/blob/main/src/provider/bn256_grumpkin.rs#L97) and [from_label_serial](https://github.com/microsoft/Nova/blob/main/src/provider/bn256_grumpkin.rs#L98) on [assert_eq](https://github.com/microsoft/Nova/blob/main/src/provider/bn256_grumpkin.rs#L101).

Thus, return value of **from_label** and **from_label_serial** should be the same type.
In **from_label_serial**, we use [`hash_to_curve`](https://github.com/microsoft/Nova/blob/main/src/provider/bn256_grumpkin.rs#L85) and [`to_affine`](https://github.com/microsoft/Nova/blob/main/src/provider/bn256_grumpkin.rs#L86).
**hash_to_curve** is method of CurveExt of [pasta](https://github.com/zcash/pasta_curves/blob/main/src/arithmetic/curves.rs#L67) and **to_affine** converts the type of CurveAffine.

To match the type, it was necessary.

I may be able to do it with macros and avoid CurveAffine requirement.

Reply:
Yes, I generalized the test without changing trait.
I would appreciate it if you could check.